### PR TITLE
Cache common Primer values

### PR DESCRIPTION
**Pull request by BlakeWilliams**
* Add `Primer::Classify::Cache` class for preloading common primer classes
* Use `Primer::Classify::Cache` when available in `Primer::Classify`
* Add benchmark tests to avoid having to give imprecise counts due to
  differences in Ruby/Rails versions
* Add CI for running benchmarks against Rails main, Ruby 2.7
* Only run simplecov if `COVERAGE=1` in your environment

This takes us from 114 allocations when not cached, down to 40 allocations when cached. There's probably a lot more work that could be done here, like catching new keys that `Classify` supports and tracking that we're caching them correctly.

Co-authored with @kenyonj

### CODE REVIEW

**Comment by BlakeWilliams** [comment]
Here's the benchmark results as well:

Screen Shot 2021-02-08 at 3 31 55 PM

File: lib/primer/classify/cache.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+          )
+
+          preload(
+            keys: Primer::Classify::DISPLAY_KEY,
+            values: [:flex, :block, :inline_block, :inline_flex, :none, :table, :table_cell]
+          )
+
+          preload(
+            keys: [Primer::Classify::COLOR_KEY, Primer::Classify::BG_KEY],
+            values: [:blue, :gray_dark, :gray, :gray_light, :red, :orange, :orange_light, :yellow, :green, :purple, :white, :pink]
```
Comment (joelhawksley) [inline lib/primer/classify/cache.rb:54]: How did you arrive on these `values` here? (And for the rest of this file?) I'd love to have a process to ensure this optimization stays up to date 😄

**Review by joelhawksley — APPROVED** [review]
🤯 Next level work here, Blake and Justin! Dang!

File: lib/primer/classify/cache.rb (hunk 1 above)
Comment (BlakeWilliams) [reply lib/primer/classify/cache.rb:54]: It was very unscientific, we went through https://primer.style/css/ and tried to find colors to fill in here 😅.

There's probably a few better ways to handle it, like maybe parsing primer itself for color variables that we could use but we didn't have time to look into anything like that.

**Pull request merged**